### ISSUE TITLE
Font size preference

### ISSUE THREAD

**Issue by cfjedimaster** [adobe/brackets#10542]
Via a conversation I had with @redmunds, I should be able to set a preferred font size for Brackets by including it in my .brackets.json:

{
    "sbruchmann.staticpreview.basepath": "/Users/raymondcamden/Dropbox/projects/cordovademos/",
    "fonts.fontSize":"22px"
}

But this does not work. I'm trying it in a project specific .brackets.json. I do have a core file as well but this preference is not set there. 

In console, I don't see anything outside of this:

Blind write attempted:  /Users/raymondcamden/Dropbox/projects/cordovademos/.brackets.json 1423243116000 null

I tried both with and without extensions.

**Comment by redmunds** [comment]
@cfjedimaster 

> Blind write attempted: /Users/raymondcamden/Dropbox/projects/cordovademos/.brackets.json 1423243116000 null

That means an attempt to update the project prefs file failed. Try shutting down Brackets, making update in another editor, and restarting Brackets.

**Comment by cfjedimaster** [comment]
I closed Brackets. I opened up .brackets.json for my project in Sublime. The change is there. I had edited in Brackets before and it saved just fine. Reopened Brackets and closed the file (thinking maybe there is an issue editing .brackets.json in Brackets itself, which would suck ;), closed it, back in Sublime confirmed it was still the new size (22px), reopened Brackets and my font hasn't changed.

**Comment by cfjedimaster** [comment]
I should add, even with .brackets.json closed in Brackets itself, the console still shows the blind write error.

**Comment by cfjedimaster** [comment]
The blind write error goes away if I reload w/o extensions, but my font size still doesn't show.

**Comment by sprintr** [comment]
It is not possible to change `font-family` or `font-size` via prefs.
CC @MiguelCastillo

**Comment by MiguelCastillo** [comment]
@redmunds I thought I had added the wiring for it, but I guess I didn't.  I will make a PR for this one sometime tonight.

Here is the code I defined the font size...  But I didn't define the change handler

``` javascript
    prefs.definePreference("fontSize",   "string", DEFAULT_FONT_SIZE + "px").on("change", function() {
        console.log("font changes", arguments);
    });
```

**Comment by MiguelCastillo** [comment]
Ooops, forgot to include the link to the source I intended to in my previous post - https://github.com/adobe/brackets/blob/master/src/view/ViewCommandHandlers.js#L491

**Comment by cfjedimaster** [comment]
Well, at least I know I'm not going crazy. :)

On Fri, Feb 6, 2015 at 1:13 PM, miguel @manchagnu <notifications@github.com>
wrote:

> Ooops, forgot to include the link to the source I intended to in my
> previous post -
> https://github.com/adobe/brackets/blob/master/src/view/ViewCommandHandlers.js#L491
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adobe/brackets/issues/10542#issuecomment-73294884>.
>



-- 
===========================================================================
Raymond Camden, Web Standards Evangelist

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by MiguelCastillo** [comment]
@cfjedimaster on second thought, I should have played it out a little longer :D

**Comment by peterflynn** [comment]
@MiguelCastillo Wait, whaa? If you change font size view View > Themes or Ctrl-+/- and then reload Brackets, the change is remembered.  It's getting stored in preferences _somewhere_ already...  A missing change handler wouldn't explain the problem @cfjedimaster is having where the pref is already in the JSON file at startup and it still gets ignored anyway...

**Comment by peterflynn** [comment]
@cfjedimaster If you adjust font size by just pressing Ctrl-+, does that change stick after reloading Brackets?  If not this may be an issue specific to your system...  (Try reloading without extensions first things first!)

**Comment by MiguelCastillo** [comment]
@peterflynn the change is remembered with CMD +/- because we are setting the size in the preferences, which automatically saves it to the file...  But if the *file* is changed, we need a change handler to get the new value and trickle that change in the UI with the new font size.

**Comment by MiguelCastillo** [comment]
@peterflynn @cfjedimaster alright, I read the problem wrong.  I somehow thought that we were talking about live updating the .brackets.json showing the new font sizes in Brackets...  You need the handler I am talking about here.

But the issue is that even after reloading Bracket, the new font size does not take effect. Let's start with your suggestion @peterflynn .  No extensions...  This issue seems odd.

Sorry about the confusion!

**Comment by cfjedimaster** [comment]
Nope, it doesn't stick at all. I tried reloading normally, and reloading
w/o extensions, and it does not stick after restarting.

On Sun, Feb 8, 2015 at 7:02 PM, Peter Flynn <notifications@github.com>
wrote:

> @cfjedimaster <https://github.com/cfjedimaster> If you adjust font size
> by just pressing Ctrl-+, does that change stick after reloading Brackets?
> If not this may be an issue specific to your system... (Try reloading
> without extensions first things first!)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adobe/brackets/issues/10542#issuecomment-73444181>.
>



-- 
===========================================================================
Raymond Camden, Web Standards Evangelist

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by MiguelCastillo** [comment]
@cfjedimaster I can reproduce it.  Without debugging the issue, the font size from the global `brackets.json` is always used when brackets loads, rather then the project `.brackets.json`.  Even when there is no font size entry in `brackets.json`.  I can debug this tomorrow.

**Comment by cfjedimaster** [comment]
To be clear, I can't get Brackets to remember my last font size, *nor* will
it respect font size set in a project .brackets.json or my global
brackets.json.

On Sun, Feb 8, 2015 at 9:47 PM, miguel @manchagnu <notifications@github.com>
wrote:

> @cfjedimaster <https://github.com/cfjedimaster> I can reproduce it.
> Without debugging the issue, the font size from the global brackets.json
> is always used when brackets loads, rather then the project .brackets.json.
> Even when there is no font size entry in brackets.json. I can debug this
> tomorrow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/adobe/brackets/issues/10542#issuecomment-73453818>.
>



-- 
===========================================================================
Raymond Camden, Web Standards Evangelist

Email : raymondcamden@gmail.com
Blog : www.raymondcamden.com
Twitter: raymondcamden

**Comment by MiguelCastillo** [comment]
@cfjedimaster @peterflynn So I found a pattern... The issue is pretty odd.

You need a .brackets.json in your project folder.  Mine looks like this

```
{
    "jscodehints.detectedExclusions": [
        "/Users/mcastillo/Projects/bit-imports/src/bit-imports.js",
        "Users/mcastillo/Projects/bit-imports/test/js/deep2",
        "/Users/mcastillo/Projects/bit-imports/test/js/pathconfig.js",
        "/Users/mcastillo/Projects/bit-imports/test/js/string.js"
    ],
    "fonts.fontSize": "15px"
}
```

First test:
1. Make sure you don't have `.brackets.json` or `brackets.json` open.
2. CMD +/-, fonts change and will get saved to `.brackets.json`, which is good.
3. Reload Brackets and settings are read from `brackets.json`.

Second test:
1. Make sure you only have `brackets.json` open in brackets.
2. CMD +/-, fonts change and will get saved to `brackets.json`, which is wrong.
3. Reload Brackets and settings are read from `brackets.json`

Third test:
1. Make sure you only have `.brackets.json` open in brackets.
2. CMD +/-, fonts change and settings will get save to `.brackets.json`, which is good.
3. Reload Brackets and settings are read from `brackets.json`.

So, what I see is that:
1. Settings are saved to whichever preferences file you have open.
2. Settings are always initially read from `brackets.json`.
3. When CMD +/- is used, settings are read `.brackets.json` if no preferences is open, otherwise whichever is open is read.  You don't need to reload Brackets to see this one to see the problem.  This one is pretty crazy!

There are a couple more nuances, but I think I described most of the issues I see.

This is really weird!  Now I that come to think of it, I do remember a bit of weirdness around this when I was first working on the font sizing stuff...  But I couldn't pin down the behavior.

**Comment by redmunds** [comment]
OK, I can reproduce problem on Win7 where editing `brackets.json` does update when `.brackets.json` exists and does *not* have font size set. But, it does update when I restart Brackets.

**Comment by andreaslarsen** [comment]
After latest update on mac cmd± doesn't seem to work

**Comment by prafulVaishnav** [comment]
@andreaslarsen Can you try reloading Brackets without Extensions ?

**Comment by andreaslarsen** [comment]
@prafulVaishnav just did, still not working

**Comment by prksingh** [comment]
@andreaslarsen Could you create a new issue for this? Your issue seems different from the issue being discussed here

**Comment by andreaslarsen** [comment]
@prksingh it resolved itself (no idea how). I'll create a new issue if it reappears.

**Comment by andreaslarsen** [comment]
Cmd+- has worked fine the last 20 days but "cmd 0" resets to 12 px. 

Reproduce:
On mac 10.10.2 with brackets 1.2.0-15697 running without extensions. 
View... -> Themes -> change the font size to e.g. 16px -> press done -> cmd 0 -> Font size changes to 12 px

**Comment by peterflynn** [comment]
@andreaslarsen That's what it's supposed to do -- you're activating the View > Reset Font Size menu item.  Please file a spinoff bug though -- the discussion thread here is about a different topic.
